src/tools/utils.ts

```diff
@@ -79,15 +79,24 @@ export async function runAndWait(context: Context, status: string, callback: (pa
   };
 }
 
+export async function captureAllFrameSnapshot(page: playwright.Page): Promise<string> {
+  const snapshots = await Promise.all(page.frames().map(frame => frame.locator('html').ariaSnapshot({ ref: true })));
```

Comment:
Let's query all frames via DOM, create frame locators for the nodes that we discover and map frame ids => locator prefixes.